web/client/utils/WidgetsUtils.js

```diff
@@ -47,14 +48,14 @@ const getConnectionList = (widgets = []) => {
 };
 
 /**
- * it checks if a number is higher of 10k and it returns a shortened version of it
+ * it checks if a number is higher than threshold and returns a shortened version of it
  * @param {number} label to parse
  * @param {number} threshold threshold to check if it needs to be rounded
  * @param {number} decimals number of decimal to use when rounding
  * @return the shortened number plus a suffix or the label is a string is passed
 */
-const shortenLabel = (label, threshold = 10000, decimals = 1)=> {
-    if (!isNumber(label) || isNumber(label) && label.toString().length < 7) {
+const shortenLabel = (label, threshold = (ConfigUtils.getConfigProp('shortenLabelThreshold') || 1000), decimals = 1) => {
```

Comment:
Hi, ususally we do not add new config properties at root level if they belongs to some particular plugin.
In this case it should be the Widget Plugin

Try to add this property in the Widget section of localConfig.json and then to pass it from the plugin to the final component (YAxisLabel). at that point you can extend the shortenLabel signature by adding the new threshold prop